modules/gdscript/tests/test_completion.h

```diff
@@ -0,0 +1,200 @@
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_COMPLETION_H
+#define TEST_COMPLETION_H
+
+#ifdef TOOLS_ENABLED
+
+#include "core/io/config_file.h"
+#include "core/io/dir_access.h"
+#include "core/io/file_access.h"
+#include "core/object/script_language.h"
+#include "core/variant/dictionary.h"
+#include "core/variant/variant.h"
+#include "gdscript_test_runner.h"
+#include "modules/modules_enabled.gen.h" // For mono.
+#include "scene/resources/packed_scene.h"
+
+#include "../gdscript.h"
+#include "tests/test_macros.h"
+
+#include "editor/editor_settings.h"
+#include "scene/theme/theme_db.h"
+
+namespace GDScriptTests {
+
+bool match_option(Dictionary expected, ScriptLanguage::CodeCompletionOption got) {
+	if (expected.get("display", got.display) != got.display) {
+		return false;
+	}
+	if (expected.get("insert_text", got.insert_text) != got.insert_text) {
+		return false;
+	}
+	if (expected.get("kind", got.kind) != Variant(got.kind)) {
+		return false;
+	}
+	if (expected.get("location", got.location) != Variant(got.location)) {
+		return false;
+	}
+	return true;
+}
+
+void to_dict_list(Variant p_variant, List<Dictionary> &p_list) {
+	if (p_variant.is_array()) {
+		Array arr = static_cast<Array>(p_variant);
+		while (arr.size() > 0) {
+			Variant E = arr.pop_front();
+			if (E.get_type() == Variant::DICTIONARY) {
+				p_list.push_back(static_cast<Dictionary>(E));
+			}
+		}
+	}
+}
+
+void test_directory(const String &p_dir) {
+	Error err = OK;
+	Ref<DirAccess> dir(DirAccess::open(p_dir, &err));
```

Comment:
```suggestion
	Ref<DirAccess> dir = DirAccess::open(p_dir, &err);
```
Cleaner IMO, and most commonly used